src/NextGen/DonationForm/Actions/ConvertDonationFormBlocksToFieldsApi.php

```diff
@@ -78,58 +85,81 @@ protected function convertTopLevelBlockToSection(BlockModel $block, int $blockIn
      * @since 0.1.0
      *
      * @throws EmptyNameException|NameCollisionException
+     *
+     * @return Node|null
      */
-    protected function convertInnerBlockToNode(BlockModel $block, int $blockIndex): Node
+    protected function convertInnerBlockToNode(BlockModel $block, int $blockIndex)
     {
         $node = $this->createNodeFromBlockWithUniqueAttributes($block, $blockIndex);
 
-        return $this->mapGenericBlockAttributesToNode($node, $block);
+        if ($node instanceof Node) {
+            return $this->mapGenericBlockAttributesToNode($node, $block);
+        }
+
+        return null;
     }
 
     /**
-     * @unreleased add blockIndex for unique field names
+     * @unreleased add blockIndex for unique field names, add filter `givewp_donation_form_block_render_{$blockName}`
      * @since 0.1.0
      *
      * @throws EmptyNameException
      * @throws NameCollisionException
+     *
+     * @return Node|null
      */
-    protected function createNodeFromBlockWithUniqueAttributes(BlockModel $block, int $blockIndex): Node
+    protected function createNodeFromBlockWithUniqueAttributes(BlockModel $block, int $blockIndex)
     {
-        switch ($block->name) {
-            case "custom-block-editor/donation-amount-levels":
+        $blockName = str_replace("custom-block-editor/", '', $block->name);
```

Comment:
@jonwaldstein My only concern on this is that we are ignoring namespaces. This can cause collisions in the future in case two distinct developers decide to create blocks with the same names. I'd keep the namespace here, either for the switch-case section or for the hook.

Reply:
@pauloiankoski that's a great observation, honestly I was thinking every block would use the same namespace because I thought we _had_ to - but since you said that i'm wondering if external blocks can and _should_ use a different namespace to avoid conflicts.  This might be the time to change `custom-block-editor` to `givewp` and test out a different namespace for a custom block.

Reply:
I totally agree with that.

Reply:
@pauloiankoski this has been updated ea1afe73

The only thing i'm questioning is if we should continue with the `{block-name}-field` convention or omit the word `-field` all together.

Examples:

- `givewp/text-field` vs `givewp/text`
- `givewp/email-field` vs `givewp/email`
- `givewp/donation-amount-field` vs `givewp/donation-amount`
- `givewp-fee-recovery/fee-amount-field` vs `givewp-fee-recovery/fee-amount`

On one hand, there could be value in keeping primitive / core fields keeping the `-field` suffix to differentiate them.
On the other hand I could be convinced that it's redundant or not necessary with the use of namespaces.


I would like to agree on a naming convention before this breaking change.

cc: @kjohnson @JasonTheAdams

Reply:
Gutenberg, for instance, has blocks simple: `core/cover`. I'd go without that redundancy and make `givewp/text` the standard. Waiting to hear from the others too tho.

Reply:
That's a good point, the main difference for us is that some of our blocks are technically fields/inputs that interact with a form which is my only hesitation.  Still though, i'm not sure that really matters when naming a block under the hood.

Reply:
@pauloiankoski i've decided to simplify the block names by omitting the suffix.  I would rather lean on namespaces and categories to identify the blocks than a naming convention for the block id itself.  Thanks for thinking through this with me.